SysArchMeter.c

```diff
@@ -0,0 +1,74 @@
+/*
+htop - SysArchMeter.c
+(C) 2021 htop dev team
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+#include "config.h"  // IWYU pragma: keep
+
+#include "SysArchMeter.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+#include "XUtils.h"
+
+
+static const int SysArchMeter_attributes[] = {HOSTNAME};
+
+static void SysArchMeter_updateValues(Meter* this, char* buffer, size_t size) {
+   static struct utsname uname_info;
+   static int uname_result;
+   static char distro[256] = {'\0'};
+   static bool loaded_data = false;
+
+   (void)this;
+
+   if(!loaded_data)
+   {
+      uname_result = uname(&uname_info);
+      FILE* fp = popen("lsb_release --id --release --codename --short", "r");
+      if(fp != NULL)
+      {
+         if(fread(distro, sizeof(*distro), sizeof(distro), fp))
+         {
+            size_t num_lines = 0;
+            char** d_split = String_split(distro, '\n', &num_lines);
+            distro[0] = '\0';
+            if(num_lines == 3) // lsb_release returns exactly 3 lines
+               snprintf(distro, sizeof(distro), "%s %s (%s)", d_split[0], d_split[1], d_split[2]);
+            String_freeArray(d_split);
+         }
+         pclose(fp);
+      }
+      loaded_data = true;
+   }
+
+   if(uname_result == 0)
+   {
+      if (strlen(distro) == 0)
+         snprintf(buffer, size, "%s %s [%s]", uname_info.sysname, uname_info.release,
+                  uname_info.machine);
+      else
+         snprintf(buffer, size, "%s %s [%s] / %s", uname_info.sysname, uname_info.release,
+                  uname_info.machine, distro);
```

Comment:
No need to line-wrap here. Better have one longer line, than arbitrarily word-wrap inside the argument list.
```suggestion
         snprintf(buffer, size, "%s %s [%s]", uname_info.sysname, uname_info.release, uname_info.machine);
      else
         snprintf(buffer, size, "%s %s [%s] / %s", uname_info.sysname, uname_info.release, uname_info.machine, distro);
```

Reply:
> No need to line-wrap here. Better have one longer line, than arbitrarily word-wrap inside the argument list.

That results in lines of 152 and 106 characters respectively:

```c
   if(uname_result == 0) {
      if (lsb_valid)
         snprintf(buffer, size, "%s %s [%s] / %s %s (%s)", uname_info.sysname, uname_info.release, uname_info.machine, distro[0], distro[1], distro[2]);
      else
         snprintf(buffer, size, "%s %s [%s]", uname_info.sysname, uname_info.release, uname_info.machine);
   }
```

Would it be better to line-wrap in groups of 3 instead?

```c
   if(uname_result == 0) {
      if (lsb_valid)
         snprintf(buffer, size, "%s %s [%s] / %s %s (%s)", 
                 uname_info.sysname, uname_info.release, uname_info.machine, 
                 distro[0], distro[1], distro[2]);
      else
         snprintf(buffer, size, "%s %s [%s]",
                 uname_info.sysname, uname_info.release, uname_info.machine);
   }
```

Reply:
There are several other lines, that are similarly long. Your idea works too.